skrf/media/physicalLine.py

```diff
@@ -0,0 +1,177 @@
+        self.f_low, self.f_high, self.f_epr_tand = f_low, f_high, f_epr_tand
+    
+    def __str__(self):
+        f=self.frequency
+        output = 'Physical line Media.  %i-%i %s.  %i points'%\
+                (f.f_scaled[0],f.f_scaled[-1],f.unit, f.npoints)
+        return output
+
+    def __repr__(self):
+        return self.__str__()
+    
+    @property
+    def ep_r_f(self):
+        '''
+        Frequency dependant relative permittivity of dielectric
+        '''
+        ep_r, tand  = self.ep_r, self.tand
+        f_low, f_high, f_epr_tand = self.f_low, self.f_high, self.f_epr_tand
+        f = self.frequency.f
+        if self.diel == 'djordjevicsvensson':
+            # compute the slope for a log frequency scale, tanD dependant.
+            m = (ep_r*tand)  * (pi/(2*log(10)))
+            # value for frequency above f_high
+            ep_inf = (ep_r - 1j*ep_r*tand - m*log((f_high + 1j*f_epr_tand)/(f_low + 1j*f_epr_tand)))
+            return ep_inf + m*log((f_high + 1j*f)/(f_low + 1j*f))
+        elif self.diel == 'frequencyinvariant':
+            return ones(self.frequency.f.shape) * (ep_r - 1j*ep_r*tand)
+        else:
+            raise ValueError('Unknown dielectric dispersion model')
+    
+    @property
+    def tand_f(self):
+        '''
+        Frequency dependant dielectric loss factor
+        '''
+        ep_r = self.ep_r_f
+        return -imag(ep_r) / real(ep_r)
+         
+    @property
+    def alpha_conductor(self):
+        '''
+        Losses due to conductor resistivity
+
+        Returns
+        --------
+        alpha_conductor : array-like
+                lossyness due to conductor losses
+
+        '''
+        A, f_A, f = self.A, self.f_A, self.frequency.f
+        return A * log(10)/20 * sqrt(f/f_A)
+    
+    @property
+    def alpha_dielectric(self):
+        '''
+        Losses due to dielectric
+
+        '''
+        ep_r, tand = real(self.ep_r_f), self.tand_f
+        f = self.frequency.f
+        return pi*sqrt(ep_r)*f/c * tand
+    
+    @property
+    def beta_phase(self):
+        '''
+        Phase parameter
+        '''
+        ep_r, f = real(self.ep_r_f), self.frequency.f
+        return 2*pi*f*sqrt(ep_r)/c
+
+    @property
+    def gamma(self):
+        '''
+        Propagation constant
+
+
+        See Also
+        --------
+        alpha_conductor : calculates losses to conductor
+        alpha_dielectric: calculates losses to dielectric
+        beta            : calculates phase parameter
+        '''
+        beta  = self.beta_phase
+        alpha = self.alpha_conductor + self.alpha_dielectric
+        return alpha + 1j*beta
+    
+    @property
+    def Z0(self):
+        '''
+        Characteristic Impedance, :math:`Z0`
+
+        .. math::
+                Z_0 = \\sqrt{ \\frac{Z^{'}}{Y^{'}}}
+
+        Returns
+        --------
+        Z0 : numpy.ndarray
+                Characteristic Impedance in units of ohms
+        '''
+        return zeros(len(self))  
```

Comment:
i dont understand this?

Reply:
I still have to finish this. This is more difficult because the impedance dispersion model depend of the geometry, and there is no geometry in this media implementation.
Maybe we should let the user provide a single value (no dispersion) or a vector, letting him compute his own dispersion.

By the way, it is required to write media.line(l, 'm', embed=True, z0=y) to build a line of impedance y which is normalized with media z0. Is there a specific reason not to have embed=True the default behaviour? I am thinking in the direction of getting the same results when plotting the line as if measuring with a VNA.

Reply:
ok. perhaps henrik might have a good idea, 

on the second point, it is a matter of taste.  i like to explicitly add connectors, so i frequently do, 
    wg.thru()**wg.line(...z0='2')**wg.thru()

even though you could just do 

    wg.line(z0='2', embed=1)

also if you are making a taper, or a filter, ie  cascading many lines of various impedances, embedding to a `system` or `port` impedance for each element is a waste of cpu.

Reply:
Ok, I got the idea. Indeed in most simulation softwares, the ports are added with their characteristic impedance. The vna ports have intrinsically their own port impedance too. The implemention now make sense to me.

Reply:
I just kept the characteristic impedance constant for time being. The media.thru() notation is now used in the example.